tests/test_decorators.py

```diff
@@ -85,3 +86,106 @@ class ShardedTestModelIDsThree(TableStrategyModel):
             pass
 
         self.assertEqual(getattr(ShardedTestModelIDsThree, 'django_sharding__database', None), 'app_shard_002')
+
+    def test_abstract_model_with_defined_manager_raises_exception_if_not_instance_of_shard_manager(self):
+        # Manager is defined and not shard model, should raise an exception
+        with self.assertRaises(ShardedModelInitializationException):
```

Comment:
Can you swap these to `assertRaisesRegexp` just to be sure that the exception you think is happening, is happening, since these are all the same class.